src/seq/mod.rs

```diff
@@ -309,9 +319,10 @@ pub trait IteratorRandom: Iterator + Sized {
     /// `choose` returning different elements. If you want consistent results
     /// and RNG usage consider using [`IteratorRandom::choose_stable`].
     fn choose<R>(mut self, rng: &mut R) -> Option<Self::Item>
-    where R: Rng + ?Sized {
+    where
+        R: Rng + ?Sized,
+    {
         let (mut lower, mut upper) = self.size_hint();
-        let mut consumed = 0;
         let mut result = None;
 
         // Handling for this condition outside the loop allows the optimizer to eliminate the loop
```

Comment:
This was added as an optimisation. Does it still make sense?

Reply:
Yes, that first block speeds up the size-hinted path by about 50%